impl/client/src/main/javascript/web/prompting/PromptPanel.js

```diff
@@ -34,1658 +33,1702 @@
-         * @param {Object}  [options]        Additional configuration options.
-         * @param {Boolean} [options.isInit] Flag indicating if submit is being executed during initialization.
-         */
-        submit: function (promptPanel, options) {
-          if (this.onSubmit) {
-            if (typeof this.onSubmit === "function") {
-              this.onSubmit(options);
-            } else {
-              Logger.warn("The onSubmit event callback is not a function");
+            if(c.param) {
+              c.param = paramDefn.getParameter(c.param.name);
             }
-          }
-        },
-
-        /**
-         * Called when the prompt-panel component's submit button is pressed (mouse-down only).
-         *
-         * @name PromptPanel#submitStart
-         * @method
-         * @param {PromptPanel} promptPanel
-         */
-        submitStart: function (promptPanel) {
-        },
-
-        /**
-         * Called when a parameter value changes.
-         *
-         * The current implementation of  WidgetBuilder#build hooks
-         * a method to the "postChange" CDF method of just built widgets
-         * that have a "parameter".
-         * This method calls its PromptPanel's "parameterChanged" method.
-         *
-         * @name PromptPanel#parameterChanged
-         * @method
-         * @param {Parameter} param
-         * @param {String} name
-         * @param {Object} value
-         */
-        parameterChanged: function (param, name, value) {
-
-          if (this.onParameterChanged) {
-            var paramCallback = this.onParameterChanged[name] ?
-                  this.onParameterChanged[name] :
-                  this.onParameterChanged[''];
-            if (paramCallback) {
-              if (typeof paramCallback === 'function') {
-                paramCallback(name, value, param);
-              } else {
-                Logger.warn("The parameterChanged callback for '" + name + "' is not a function");
+          });
+
+          this._focusedParam = focusedParam;
+        }
+
+        this.init(noAutoAutoSubmit);
+      }
+    },
+
+    /**
+     * Removes a set of components determined by the ParameterDefinitionDiffer#diff
+     *
+     * @name PromptPanel#_removeComponentsByDiff
+     * @method
+     * @param {JSON} toRemoveDiff The group of paramters which need to be removed
+     */
+    _removeComponentsByDiff: function(toRemoveDiff) {
+      var toRemove = [];
+      for(var groupName in toRemoveDiff) {
+        var removeWrap = toRemoveDiff[groupName];
+        var params = removeWrap.params;
+
+        for(var i = 0; i < params.length; i++) {
+          var param = params[i];
+          var component = _getComponentByParam.call(this, param, true); // get component panel by param
+          if(component != null) {
+            toRemove.push(component);
+
+            // removes the component from the group panel and also removes
```

Comment:
Is it ok to set `this.isEnableSubmitButton` to `true` when there is no submit button?

Reply:
There shouldn't be a problem with this because the html for the button is only added later having this variable into consideration. The code is on the file _ScopedPentahoButtonComponent.js_ and function _registerSubmitClickEvent_